### ISSUE TITLE
Lower response latency

### ISSUE THREAD

**Issue by jtrakk** [GenieFramework/Genie.jl#361]
I'm trying to evaluate some options where response latency is relevant. In a "hello world" example, I'm getting about 16ms from Genie server on localhost. A [hello world example with Python's Werkzeug](https://www.palletsprojects.com/p/werkzeug/) takes ~5ms on my machine and others are faster in [TechEmpower latency benchmarks](https://www.techempower.com/benchmarks/#section=data-r20&hw=ph&test=plaintext) (click the "Latency" tab).

```jl
using HTTP

HTTP.serve() do request::HTTP.Request
   request
   request.method
   HTTP.header(request, "Content-Type")
   HTTP.payload(request)
   try
       return HTTP.Response("Hello")
   catch e
       return HTTP.Response(404, "Error: $e")
   end
end
```

takes 3.8ms.

I would like to have a low-latency response comparable to other fast frameworks.

**Comment by essenciary** [comment]
Thanks - how did you run the benchmarks and what is the Genie bench-marked code?

**Comment by jtrakk** [comment]
Code is

```jl
using Genie, Genie.Router
route("/hello") do; "Hello world" end
up()
```

Benchmarked with [hyperfine](https://github.com/sharkdp/hyperfine).

```
% hyperfine  -- 'curl 127.0.0.1:8000/hello'
Benchmark #1: curl 127.0.0.1:8000/hello
  Time (mean ± σ):      15.2 ms ±   0.3 ms    [User: 2.3 ms, System: 1.5 ms]
  Range (min … max):    14.9 ms …  16.3 ms    154 runs
```

**Comment by essenciary** [comment]
@jtrakk Thanks for the details, I'll run some tests myself and do some profiling. 

What I can say for sure is that unlike say HTTP.jl (which is the app server that Genie is also using internally) Genie runs in 2 modes: dev (which is the default) and is optimized for developer productivity (eg hot code reload, automatic Revise of files, logging etc) and prod, which is streamlined to run in production and disables some of the dev features. 

First step would be to run the benchmark in production mode - just pass `GENIE_ENV=prod` to the julia process running the app. That should make a good difference. But like I said, I'll run some benchmarks myself, surely things can be further improved.

**Comment by essenciary** [comment]
Also, can you please share the output of `julia> versioninfo()`

**Comment by jtrakk** [comment]
```
julia> versioninfo()
Julia Version 1.5.4
Commit 69fcb5745b* (2021-03-11 19:13 UTC)
Platform Info:
  OS: Linux (x86_64-unknown-linux-gnu)
  CPU: AMD Ryzen 9 3900XT 12-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, znver2)
Environment:
  JULIA_PKGDIR = 
```

**Comment by jtrakk** [comment]
I got similar result with `GENIE_ENV=prod`.

**Comment by essenciary** [comment]
Bummer - I'll run some tests but it would be awesome is somebody passionate about performance tweaking would take this, not my strength.

**Comment by jtrakk** [comment]
I'm not that familiar with the profiling tools but I gave it a shot. I edited `setup_http_handler` to use `@profile handle_request(req, res)` instead of `handle_request(req, res)` and made a flamegraph that showed a lot of time being spent in `route_request` and especially in the line

```jl
req, res, params.collection = f(req, res, params.collection)
```
which it said used "runtime dispatch". But honestly I may be reading it completely wrong, especially with the use of `Distributed` and the long-running process.
